Partial rebuild preliminary work

dsavitskiy opened the pull request.

dsavitskiy committed:
feat(nexus): improving nexus fault injection

Signed-off-by: Dmitry Savitskiy <dmitry.savitskiy@datacore.com>

dsavitskiy committed:
feat(nexus): refactoring device retire code structure

Signed-off-by: Dmitry Savitskiy <dmitry.savitskiy@datacore.com>

dsavitskiy commented:
bors try

blaisedias commented:
@tiagolobocastro  @avishnu, should we defer merging this PR until after we branch for the release?

dsavitskiy commented:
It's some preliminary work needed for partial rebuild. It can be left out for the release. I created this PR to run tests to check if anything is broken since last time I touched my partial rebuild code.

dsavitskiy committed:
chore(libs): updating to latest libnvme

Signed-off-by: Dmitry Savitskiy <dmitry.savitskiy@datacore.com>

dsavitskiy committed:
chore(ci): updating to latest rustc and nix

Following dependencies updated:
rustc: 1.56.1 -> 1.66 stable
cargo: 1.58.0-nightly -> 1.68.0-nightly (2023-01-04)
rustfmt: 1.4.38-nightly -> 1.5.1-nightly (2023-01-07)
clippy: 0.1.58-nightly -> 0.1.68-nightly (2023-01-07)

nix: pre-21.11 -> 22.11
fio: 3.26 -> 3.33
nvme cli: 1.14 -> 2.2.1

Updating to the latest Rust required updating to the latest Nix packages
(version 22-11). That required bumping fio and nmve cli versions.

Signed-off-by: Dmitry Savitskiy <dmitry.savitskiy@datacore.com>

dsavitskiy committed:
feat(tests): improving file compare utility functions

Signed-off-by: Dmitry Savitskiy <dmitry.savitskiy@datacore.com>

dsavitskiy committed:
feat(nexus): implemeting log-based rebuild

Signed-off-by: Dmitry Savitskiy <dmitry.savitskiy@datacore.com>
